src/commonMain/kotlin/fr/acinq/lightning/channel/WaitForAcceptChannel.kt

```diff
@@ -79,6 +117,33 @@ data class WaitForAcceptChannel(
         }
     }
 
+    private fun createFundingTx(fundingPubkeyScript: ByteVector): Pair<Transaction, Satoshi> {
+        val txIn = init.fundingInputs.inputs.map { i -> TxIn(i.outpoint, 0) }
+        val unsignedTx = when (val changePubKey = init.fundingInputs.changePubKey) {
+            null -> Transaction(2, txIn, listOf(TxOut(init.fundingAmount, fundingPubkeyScript)), currentBlockHeight.toLong())
```

Comment:
If don't understand why we compute and substract fees only when a change pubkey is provided ?

Reply:
Because if we don't have a change pubkey, we cannot create a change output at all, we have no instructions on where to send the funds?

Reply:
The reason the change pubkey is optional is that when you're doing a swap-in, you'll want to use all the swapped-in funds and don't want a change output. This also simplifies the wallet which won't have to track as many addresses since unspent funds are generally fully consumed by a channel open without generating new unspent funds.

Reply:
Then fees could be subtracted from the amount being sent. This is what bitcoin core does for example.
What seems to be inconsistent here is how fees are handled: caller should select utxos, compute fees and subtract them from `fundingAmount` if they don't include a change pubkey ? how would since work if inputs are combined with other external inputs ?

Reply:
I removed the change pubkey entirely in https://github.com/ACINQ/lightning-kmp/pull/365/commits/1b61102ac777eafb33d5b1aa2914fb5fc942a3b3 for simplicity.